src/bin/day03.rs

```diff
@@ -0,0 +1,270 @@
+// use aoc2023::prelude::*;
```

Comment:
BEFORE MERGING:
* Introduce a `PartNumberIterator` to consolidate part number parsing logic
* Introduce `Grid::adjacent_tiles(&self, row: usize, col: usize)` to consolidate that logic